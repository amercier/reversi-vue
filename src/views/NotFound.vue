<template>
  <main class="not-found">
    <h1 class="not-found-title">Oooooops!</h1>
    <h2 class="not-found-subtitle">404</h2>
    <p class="not-found-paragraph">
      The link is broken or the page has been moved. Try these pages instead:
    </p>
    <Nav
      class="not-found-paragraph"
      link-class="not-found-link"
      :links="[['/', 'Game'], ['/about', 'About']]"
    />
  </main>
</template>

<script>
import Nav from '../components/Nav.vue';

export default {
  name: 'NotFound',
  components: {
    Nav,
  },
};
</script>

<style lang="scss">
.not-found {
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: 20rem;
  margin: 0 auto;
  text-align: center;
  line-height: 2;
  min-height: calc(100vh - #{$HEADER_HEIGHT});

  &-title {
    font-size: 3rem;
  }

  &-subtitle {
    margin: 1rem 0;
    font-size: 2rem;
  }

  &-paragraph {
    margin-bottom: 2rem;
  }

  &-link {
    text-decoration: underline;
  }
}
</style>
