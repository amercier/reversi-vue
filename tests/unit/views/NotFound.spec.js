import { shallowMount } from '@vue/test-utils';
import NotFound from '@/views/NotFound.vue';

describe('NotFound.vue', () => {
  it('renders without crashing', () => {
    expect(() => shallowMount(NotFound)).not.toThrow();
  });

  it('renders its components', () => {
    const about = shallowMount(NotFound);
    expect(about.element).toMatchSnapshot();
  });
});
