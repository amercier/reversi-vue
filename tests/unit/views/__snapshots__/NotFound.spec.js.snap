// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound.vue renders its components 1`] = `
<main
  class="not-found"
>
  <h1
    class="not-found-title"
  >
    Oooooops!
  </h1>
   
  <h2
    class="not-found-subtitle"
  >
    404
  </h2>
   
  <p
    class="not-found-paragraph"
  >
    
    The link is broken or the page has been moved. Try these pages instead:
  
  </p>
   
  <nav-stub
    class="not-found-paragraph"
    linkclass="not-found-link"
    links="/,Game,/about,About"
  />
</main>
`;
